roofit/xroofit/inc/RooFit/xRooFit/xRooNode.h

```diff
@@ -156,7 +156,7 @@ class xRooNode : public TNamed, public std::vector<std::shared_ptr<xRooNode>> {
          auto &&out = std::vector<std::shared_ptr<xRooNode>>::const_iterator::operator*();
          if (out->get() && out->empty())
             out->browse();
-         return out;
+         return std::move(out);
```

Comment:
```suggestion
         out;
```
The `std::move()` on return values is unnecessary (it can even be bad for performance because it prevents the compiler from being able to do copy elision).

Reply:
added comment about apparent need for this, according to cppcheck